### PR TITLE
Added the ability to disable autoupdates

### LINKED ISSUE #2614: Allow disabling autoupdate at build-time
**Describe the bug**

In some deployment systems, flyctl will not be able to update itself, yet it tries to. Nixpkgs is one of those that I know of.
Currently this is worked around by patching the default in the source (https://github.com/NixOS/nixpkgs/pull/244494/commits/3dcfe643700283f591f61c57ffb14e0a96fa9c07) but that's not a great solution long-term.

A compile option, so this could be overwritten from go's flags would be very welcome.

### PR DESCRIPTION
**Pull request by billyb2**
### Change Summary

Added a variable to disable autoupdates at build time

What and Why:

Fixes #2614 

How:

Related to:

---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team
- [x] n/a

### CODE REVIEW

File: internal/update/update.go [hunk 1]
```diff
@@ -32,8 +32,16 @@ type Release struct {
 	Timestamp   time.Time `yaml:"timestamp"`
 }
 
+// For disabling auto updates at build time
+var disableAutoUpdate = "false"
+
 // Check reports whether update checks should take place.
 func Check() bool {
+	fmt.Println(disableAutoUpdate)
```
Comment (dangra) [inline internal/update/update.go:40]: extra debug line

File: internal/update/update.go [hunk 2]
```diff
@@ -32,8 +32,15 @@ type Release struct {
...
 }
 
+// For disabling auto updates at build time
+var disableAutoUpdate = "false"
+
 // Check reports whether update checks should take place.
 func Check() bool {
+	if disableAutoUpdate == "true" {
+		return false
+	}
```
Comment (dangra) [inline internal/update/update.go:42]: this test can be blended into the `switch` statement below and would be more idiomatic

Comment (billyb2) [reply internal/update/update.go]: 👍🏾

**Review by dangra — APPROVED** [review]

**Comment by michaeldwan** [comment]
Hey before we merge this, what's the use case? We're not going to do custom builds for different package managers. Is this for the nix people?

**Comment by billyb2** [comment]
Exactly that. Some environments don't do well with binaries updating themselves, and having a general way for people packaging flyctl to disable it is good IMO, especially considering it's just a few lines of code. I can remove the addition to the Makefile, since just having the variable is enough.

**Pull request closed without merging**